server/graphql/v2/mutation/IndividualMutations.ts

```diff
@@ -0,0 +1,89 @@
+import bcrypt from 'bcrypt';
+import express from 'express';
+import { GraphQLBoolean, GraphQLNonNull, GraphQLString } from 'graphql';
+import { GraphQLDateTime } from 'graphql-scalars';
+
+import activities from '../../../constants/activities';
+import TwoFactorAuthLib from '../../../lib/two-factor-authentication';
+import models from '../../../models';
+import { checkRemoteUserCanUseAccount } from '../../common/scope-check';
+import { Unauthorized } from '../../errors';
+import { Individual } from '../object/Individual';
+
+const individualMutations = {
+  setChangelogViewDate: {
+    type: new GraphQLNonNull(Individual),
+    description: 'Update the time which the user viewed the changelog updates. Scope: "account".',
+    args: {
+      changelogViewDate: {
+        type: new GraphQLNonNull(GraphQLDateTime),
+      },
+    },
+    resolve: async (_, { changelogViewDate }, req) => {
+      checkRemoteUserCanUseAccount(req);
+
+      const user = await req.remoteUser.update({ changelogViewDate: changelogViewDate });
+      return user.getCollective();
+    },
+  },
+  setNewsletterOptIn: {
+    type: new GraphQLNonNull(Individual),
+    description: 'Update newsletter opt-in preference. Scope: "account".',
+    args: {
+      newsletterOptIn: { type: new GraphQLNonNull(GraphQLBoolean) },
+    },
+    resolve: async (_, { newsletterOptIn }, req) => {
+      checkRemoteUserCanUseAccount(req);
+
+      const user = await req.remoteUser.update({ newsletterOptIn });
+      return user.getCollective();
+    },
+  },
+  setPassword: {
+    type: new GraphQLNonNull(Individual),
+    description: 'Set password to Individual. Scope: "account". 2FA.',
+    args: {
+      password: {
+        type: new GraphQLNonNull(GraphQLString),
+        description: 'The password to set.',
+      },
+      currentPassword: {
+        type: GraphQLString,
+        description: 'The current password (if any) to confirm password change.',
+      },
+    },
```

Comment:
This mutation should also be rate limited.

Reply:
What would we be preventing by rate limiting this?

Reply:
If an attacker gains access to a valid token to use with this mutation (password reset, or a session token), they would be able to brute force a password change, right?